src/main/java/fi/aalto/cs/apluscourses/utils/LongIdObject.java

```diff
@@ -0,0 +1,28 @@
+package fi.aalto.cs.apluscourses.utils;
+
+/**
+ * A class whose objects' equivalence is determined based on a long-valued ID field.
+ */
+public class LongIdObject {
```

Comment:
could you please elaborate a bit on the purpose of this class, but first check if [this](https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html) could have a use here?

Reply:
We have some model classes (`Exercise`, `Group`, else?) whose `equals(Object)` method should(?) be like (an example for `Group`)
```java
@Override
public boolean equals(Object obj) {
  return obj instanceof Group && ((Group) obj).getId() == getId();
}
```
(I realized I should override `equals(Object)` when I was writing tests and wanted to assert if a `Group` object returned by some method was equal to one I had used in the test. If `equal` is not overridden, reference equality is applied, but I was unsure if the object instance returned by that method was the same or not, in that case, it was unimportant implementation detail whether the same instance is used or not as long as they represent the same group.) 

I wrote this class to avoid writing similar `equals(Object)` and `hashcode()` methods to multiple classes. So the purpose of this class is/was to reduce code duplication.

However, now when I think it, there is a caveat in this solution because `a.equals(b)` (where `a` and `b` are e.g. `Group`s) only returns true if they have the exact same runtime type (because of `obj.getClass() == getClass()` clause in `equals(Object obj)`). On the other hand, that is needed because we don't want a group to equal to an exercise having the same `id` but, on the other hand, consider for example:
```java
Group group = new Group(7, Collections.emptyList());
Group sameGroup = Mockito.spy(group);
Assert.assertEquals(group, sameGroup);
```
which fails because the two objects have different runtime types. So, the conventional way to implement `equals(Object)` and `hashcode()` in `Group` class (as well as other such classes like `Exercise`, separately) is perhaps the way to go.

I'll remove this class.